llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -435,6 +435,55 @@ def IREELinalgExt_TopkOp : IREELinalgExt_Op<"topk",[
   }];
 }
 
+// TODO: is pack the right name? After all we are re-arranging the tensor.
+def IREELinalgExt_PackOp : IREELinalgExt_Op<"pack",[
+  DeclareOpInterfaceMethods<LinalgExtInterface>,
+  DeclareOpInterfaceMethods<TilingInterface,
+    ["getIterationDomain",
+     "getLoopIteratorTypes",
+     "generateScalarImplementation"]>,
+  DeclareOpInterfaceMethods<MemoryEffectsOpInterface>, 
+  SingleBlockImplicitTerminator<"::mlir::iree_compiler::IREE::LinalgExt::YieldOp">,
+]>{
+  let summary = "pack operation";
+  let description = [{
+    Expand the input tensor based on the access pattern provided by the affine
+    map.
+  }];
+  
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
+                       AffineMapAttr:$packMap);
```

Comment:
Ideas for replacing the pack map with attributes:
* Add a n-block argument, for example `{ 0, 32, 32 }`, for blocking `NHW` into `NH'W'hw` with `h = 32` and `H' = H/n` (same for W).
* Add a transpose argument, for example `[0, 2, 1]`, for transposing the last two dimensions.

I'm not sure how the blocking args would play with the transpose args (ie. do we transpose than block or block then transpose?). For example, is `#block {32, 16}, #transpose {1, 0}` block size 16 for dim 1 before or after the transpose?

Reply:
I'd like to keep the size of the inner block mixed static and dynamic, like how offsets, sizes and strides are specified in extract and insert slices. So the op could be something like 

```
%1 = linalg_ext.pack %0 inner_tile_dims [%block_0, %block_1] : tensor<?x?xf32> to tensor<?x?x?x?xf32>
```

I am assuming when you mean transpose you mean the transpose for the inner block dimensions. For that a list of integer attributes that specify the transpose would be sufficient.

Reply:
Also wondering if we could change the implementation to not use `AffineMapAttr` to start with. We can have helper functions that build the `AffineMap` if needed (like it is done for expand/collapse_shape ops. It does make the implementation of lowering to scalar simpler (as I see below), but I have found `AffineMap` to be too cumbersome to use.

Reply:
> 

To give you more context these are the blocking we would like to support: https://github.com/iree-org/iree/pull/10481#discussion_r976032524. Transposing the inner block dimensions seems sufficient.

Reply:
And yes we want to drop the affineMap!